src/lib/storage/storage_manager.rb

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yastx"
+require "storage"
+
+module Yast
+  module Storage
+    #
+    # Singleton class for the libstorage object.
+    #
+    # You can simply use StorageManager.instance to use it and create a
+    # libstorage instance if there isn't one yet; this will use common default
+    # parameters for creating it.
+    #
+    # If you need special parameters for creating it, you can use
+    # StorageManager.create_instance with a custom ::Storage::Environment
+    # (notice the difference between the global ::Storage namespace which is
+    # libstorage and this Yast::Storage namespace).
+    #
```

Comment:
just for the record - from IRC #yast:

[13:34:09] <ancorgs> HuHa: most relevant things commented by jreidinger already. I agree in all his suggestions. After fixing all that, the only thing that would remain "smelly" to me is the StorageManager class, I'll try to come up with a more constructive idea for it
[13:34:47] <HuHa> ancorgs: arvin and I had talked a lot about this
[13:35:02] <HuHa> this is the thing that makes the whole thing usable
[13:35:25] <HuHa> because only with a singleton you don't have to pass that storage object around as a parameter everywhere
[13:35:41] <HuHa> it might be used in other parts of the code, too
[13:35:49] <HuHa> and we certainly don't want to do the probing several times
[13:36:04] <HuHa> and we need to be able to set it up differently to fake storage things
[13:36:35] <HuHa> so we need to be able to initialize it from someplace else with a different ::Storage::Environment
[13:38:35] <ancorgs> HuHa: I see. I guess I need to read a little bit about ::Storage::Environment first :-)
[13:39:19] <HuHa> ancorgs: the nice thing about this approach is that you don't need to do exactly that ;-)
[13:39:23] <arvin> please review https://github.com/openSUSE/libstorage-bgl-eval/pull/25
[13:39:33] <HuHa> only if you have special requirements, you need to take care of that Environment
[13:39:37] <HuHa> in most cases, you don't
[13:41:10] <ancorgs> HuHa: well, maybe I don't need to do it as a class user, but I'll need to do it as a class code-reviewer :-)
[13:41:27] <HuHa> that's why we are talking to each other ;-)